### ISSUE TITLE
Issue with Karma: A unique name is required for a Tool Definition thrown

### ISSUE THREAD

**Issue by jwaldrip** [fraywing/textAngular#1516]
How do I fix this?

```
Chrome 58.0.3029 (Mac OS X 10.12.4) Service: speechToTextTaPluginService register should add the plugin the taRegisterTool FAILED
    textAngular Error: A unique name is required for a Tool Definition thrown
```

**Comment by pbassut** [comment]
You're registering two tools with the same name. Which is not allowed.

**Comment by pbassut** [comment]
In which point are you getting this error?

**Comment by jwaldrip** [comment]
When running my tests with webpack. I get about 700 of them, did not happen before moving to it. Is there a way to disable this check?

**Comment by ShareinSK** [comment]
I am learning test my angular app using karma jasmine.
This is my spec file

    describe('Controller', function() {
      beforeEach(module('AngularApp'));
  
    it('currentView',
            inject(function ($controller, $rootScope) {
                var $scope = $rootScope.$new();
                $controller('sampleController', {
                    $scope: $scope
                });
                expect($scope.currentView).toBeDefined();
            
            }));
    it('defaultMsg',
            inject(function ($controller, $rootScope) {
                var $scope = $rootScope.$new();
                $controller('sampleController', {
                    $scope: $scope
                });
                expect($scope.defaultMsg).toBeTruthy();
            }));
      });

When executing the above file I am getting the following error.

        Chromium 58.0.3029 (Ubuntu 0.0.0) Controller defaultMsg FAILED
        textAngular Error: A unique name is required for a Tool Definition thrown

But if I remove the defaultMsg then it is running correctly.

**Comment by Pirozek** [comment]
Also experiencing this problem, ES6, Webpack and Karma tests

**Comment by edgar0011** [comment]
no answer?

**Comment by martiikr** [comment]
Happens for me as well. The second test in my project fails.

I can see there is a fix in the textAngular tests:
 https://github.com/textAngular/textAngular/blob/ff8e48087f780d30f54e77b06f09e0b85f9517e9/src/main.js#L5

I tried it in my setup but no luck. What would be the best way to work around that problem?

**Comment by bogdan-panteleev-mifort** [comment]
Yes, I have faced this problem too. It happens, when I have more than 1 `it` function in my test script

**Comment by bogdan-panteleev-mifort** [comment]
To work around this problem I have removed `|| taTools.hasOwnProperty(name)` from the if-condition, which throws this error. It is ok now for me to test my app

**Comment by fidaurhaq** [comment]
> To work around this problem I have removed `|| taTools.hasOwnProperty(name)` from the if-condition, which throws this error. It is ok now for me to test my app

@bogdan-panteleev-mifort where did you remove this statement from?

**Comment by bogdan-panteleev-mifort** [comment]
> > To work around this problem I have removed `|| taTools.hasOwnProperty(name)` from the if-condition, which throws this error. It is ok now for me to test my app
> 
> @bogdan-panteleev-mifort where did you remove this statement from?

@fidaurhaq , In my version of package it is in line 57 in function called ''registerTextAngularTool". 
You just can try to find this string ('textAngular Error: A unique name is required for a Tool Definition'), then you will find the place, where to change

**Comment by fidaurhaq** [comment]
@bogdan-panteleev-mifort thanks! I found it.

I don't like updating the src code tho, will look for another workaround and keep this thread updated if I find one.

**Comment by fidaurhaq** [comment]
I figured out a workaround. The module that has textAngular injected into it, can be loaded using beforeAll if you're using Jasmine. This will make sure that textAngular is not re-instantiated with each test. 

So If you have an app
`angular.module('app', ['textAngular'])`

then in your tests, you will have:

`beforeAll(module('app'));`

Be-careful with this approach since you may leak state between specs.

**Comment by bogdan-panteleev-mifort** [comment]
Ok, thanks for your solution

**Comment by fidaurhaq** [comment]
I have since realized the solution i proposed doesn't always work, we've created a local copy of textAngular with @bogdan-panteleev-mifort 's fix and use that just for the tests, otherwise we load up the dependency from node from our actual app
